src/main/java/com/example/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,148 @@
+package com.example.service.impl;
+
+import com.example.dto.order.CreateOrderRequestDto;
+import com.example.dto.order.OrderItemResponseDto;
+import com.example.dto.order.OrderResponseDto;
+import com.example.dto.order.UpdateOrderRequestDto;
+import com.example.exception.EntityNotFoundException;
+import com.example.mapper.OrderItemMapper;
+import com.example.mapper.OrderMapper;
+import com.example.model.CartItem;
+import com.example.model.Order;
+import com.example.model.OrderItem;
+import com.example.model.ShoppingCart;
+import com.example.model.User;
+import com.example.repository.cartitem.CartItemRepository;
+import com.example.repository.order.OrderItemRepository;
+import com.example.repository.order.OrderRepository;
+import com.example.repository.shoppingcartrepository.ShoppingCartRepository;
+import com.example.service.OrderService;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final OrderRepository orderRepository;
+    private final CartItemRepository cartItemRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    @Override
+    public List<OrderResponseDto> getOrderHistory(User user, Pageable pageable) {
+        return orderRepository.findOrdersByUser(user, pageable).stream()
+            .map(orderMapper::toDto)
+            .toList();
+    }
+
+    @Override
+    public Set<OrderItemResponseDto> getOrderItemsByOrderId(User user, Long orderId) {
+        Order order = getOrderByIdAndUser(user,orderId);
+
+        return order.getOrderItems().stream()
+            .map(orderItemMapper::toDto)
+            .collect(Collectors.toSet());
```

Comment:
Mapstruct supports mapping collections from the box

Reply:
done

Reply:
Yes